src/domains/domd/domd_cfg_spider.c

```diff
@@ -283,7 +283,8 @@ static ssize_t get_ipl_image_size(void* image_info, uint64_t* size)
 }
 
 struct xen_domain_cfg domd_cfg = {
-    .machine_dt_compat = "renesas,r8a779f0",
+    .machine_dt_compat = (char *[]){"renesas,r8a779f0"},
```

Comment:
This is horrible. Also you are moving `const` specifier.

Reply:
sorry, I don't understand your comment
it is done in the same way as it was done for other string array fields in this structure
but I set values in place without using an extra array

Reply:
1. String constant have type `const char*`. You are removing `const` qualifier here. This is the open road to issues in the future.
2. Why do you that cast in the first place?

Reply:
1. Okay, I fixed it for the machine dt field.
 
2. You should read about array to pointer conversion. To understand this concept, in short, `{"STR"}` is treated by the compiler as `char *`. For a more detailed explanation, refer to paragraph 3 of section `6.3.2.1 Lvalues, arrays, and function designators` in the C99 standard.

So, for suppressing compiler warnings I need to apply an explicit cast.

Reply:
String literals are stored in `ro` memory. Watch this:

```
$ cat ~/tmp/test.c
struct xen_domain_cfg {
	char **machine_dt_compat;
};

struct  xen_domain_cfg domcfg = {
	.machine_dt_compat = (char *[]){"renesas,r8a779f0", "aaa"},
};

int main(int argc, char **argv)
{
	domcfg.machine_dt_compat[0][0] = 'a';
	return 0;
}
```

```
$ gcc ~/tmp/test.c       
$ ./a.out  
[1]    45949 segmentation fault (core dumped)  ./a.out
```

Your cast conceals this issue. This is why I telling you that I do not like that cast.  It is very easy to man an error.

Reply:
I know that string literals is stored in RO
like I wrote before I fixed it, just forgot to push changes